diff --git a/src/platform/linux/audio.cpp b/src/platform/linux/audio.cpp
index 9d1edb9..d915f2b 100644
--- a/src/platform/linux/audio.cpp
+++ b/src/platform/linux/audio.cpp
@@ -84,6 +84,6 @@ namespace platf {
       .tlength = uint32_t(-1),
       .prebuf = uint32_t(-1),
       .minreq = uint32_t(-1),
-      .fragsize = uint32_t(frame_size * channels * sizeof(float))
+      .fragsize = uint32_t(frame_size * 8 * channels * sizeof(float))
     };
 
